packages/autocomplete-plugin-redirect-url/src/createRedirectUrlPlugin.ts

```diff
@@ -0,0 +1,187 @@
+    templates: defaultTemplates,
+    onRedirect: defaultOnRedirect,
+    ...options,
+  };
+}
+
+function getRedirectData({ state }) {
+  const redirectUrlPlugin = state.context
+    .redirectUrlPlugin as RedirectUrlPluginData;
+  return (redirectUrlPlugin?.data ?? []) as RedirectUrlItem[];
+}
+
+export function createRedirectUrlPlugin<TItem extends BaseItem>(
+  options: CreateRedirectUrlPluginParams<TItem> = {}
+): AutocompletePlugin<RedirectUrlItem, undefined> {
+  const { transformResponse, templates, onRedirect } = getOptions(options);
+
+  function createRedirects({ results, source, state }): RedirectUrlItem[] {
+    const redirect: RedirectUrlItem = {
+      sourceId: source.sourceId,
+      urls: results
+        .map((result) => transformResponse(result))
+        .filter((url) => url !== undefined),
+    };
+
+    const redirects: RedirectUrlItem[] =
+      state.context.redirectUrlPlugin?.data ?? [];
+    const existingRedirectIndex = redirects.findIndex(
+      (r) => r.sourceId === source.sourceId
+    );
+
+    if (existingRedirectIndex !== -1) {
+      if (redirect.urls.length === 0) {
+        redirects.splice(existingRedirectIndex, 1);
+      } else {
+        redirects[existingRedirectIndex] = redirect;
+      }
+    } else if (redirect.urls.length > 0) {
+      redirects.push(redirect);
+    }
+
+    return redirects;
+  }
+
+  return {
+    name: 'aa.redirectUrlPlugin',
+    subscribe({ onResolve, setContext }) {
+      onResolve(({ results, source, state }) => {
+        setContext({
+          ...state.context,
+          redirectUrlPlugin: {
+            data: createRedirects({ results, source, state }),
+          },
+        });
+      });
+    },
+    reshape({ state, sourcesBySourceId }) {
+      const redirects =
+        (state.context.redirectUrlPlugin as RedirectUrlPluginData)?.data ?? [];
+
+      redirects.forEach((redirect) => {
+        const source = sourcesBySourceId[redirect.sourceId];
+        if (source === undefined) {
+          return;
+        }
+
+        sourcesBySourceId[redirect.sourceId] = {
+          ...source,
+          getItems: () =>
+            source.getItems().filter((item) => {
+              if (source.getItemInputValue!.__default) {
+                warn(
+                  false,
+                  `The source ${source.sourceId} does not have a getItemInputValue function. It's required to be able to filter out the redirect item.`
+                );
+                return true;
+              }
+
+              const itemInputValue = source.getItemInputValue!({
+                item,
+                state,
+              });
+
+              if (itemInputValue === undefined) {
+                warn(
+                  false,
+                  `The source ${source.sourceId} does not return a string from the getItemInputValue function. It's required to be able to filter out the redirect item.`
+                );
+                return true;
+              }
```

Comment:
I didn't see this specificity listed in the RFC. Could the item value be inferred instead?

Reply:
This was the methodology we had in mind writing the RFC. What do you see as alternative? Checking every key of the object maybe (recursively?)?

Reply:
You're right it doesn't seem efficient and reliable if we try to find it ourselves. I'll just suggest adding a link to the API ref so it's easier for builders.